### DISCUSSION TITLE
Add buffer completion in : substitute command

### DISCUSSION THREAD

**Discussion by LamprosPitsillos** [hrsh7th/nvim-cmp#970] [Q&A]
Is it possible to add completion to the `:s` command?
Example:
```
...
Hello world!
Hey there!
...

```
`:%s/He<tab>` and it shows buffer completion menu "hello, hey.."

**Comment by Shougo** [comment]
I think it is possible with command line completion.

**Comment by LamprosPitsillos** [comment]
Do we have any updates on this?

**Comment by fitrh** [comment]
You can add `bufer` source to `setup.cmdline(':')`

```lua
cmp.setup.cmdline(":", {
    completion = { keyword_length = 1 },
    sources = {
        { name = "cmdline" },
        { name = "buffer", option = { keyword_length = 1 } },
        { name = "path" },
    },
})
```

**Reply by fitrh** [reply]
By default, the `buffer` source set the `keyword_length` to 3
